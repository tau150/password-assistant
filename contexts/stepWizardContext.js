import * as React from 'react'

export const StepWizardContext = React.createContext()

export function useStepsWizard(){
  const context = React.useContext(StepWizardContext)
  if(!context){
    throw new Error('useStepsWizard must be used within a Wizard')
  }
  return React.useContext(StepWizardContext)
}