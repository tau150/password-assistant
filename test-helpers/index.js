
import * as React from 'react'
import {   render as rtlRender } from '@testing-library/react'
import { IntlProvider } from 'react-intl';
import { ChakraProvider } from "@chakra-ui/react"

import theme from '../styles/theme';
import intlENmessages from '../content/locale/en.json'

function AppProviders({children}){
  return (
    <IntlProvider
      locale='en'
      messages={intlENmessages}
      onError={() => null}>
      <ChakraProvider theme={theme}>
        {children}
      </ChakraProvider>
    </IntlProvider>
  )
}


export function render(ui, ...renderOptions){
  const returnValue = {
    ...rtlRender(ui, {wrapper: AppProviders, ...renderOptions}),
  }
  return returnValue
}
