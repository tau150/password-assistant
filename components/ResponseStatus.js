import * as React from 'react'
import PropTypes from 'prop-types'
import { Flex, Box } from '@chakra-ui/react'
import { useIntl } from 'react-intl'

import { status as statusConst } from '../constants/ui'
import { useStepsWizard } from '../contexts/stepWizardContext'
import WizardFooter from './Wizard/WizardFooter'
import StatusCard from './ui/StatusCard'

function ResponseStatus({status}){
  const intl = useIntl()
  const { handlePrevStep, setCurrentStep } = useStepsWizard()

  const title = status === statusConst.SUCCESS  ?
    intl.formatMessage({ id: 'passwordCreated', defaultMessage: 'Your Password was created' }) :a
    intl.formatMessage({ id: 'passwordError', defaultMessage: 'There was an error' })
  const body = status === statusConst.SUCCESS  ?
   intl.formatMessage({ id: 'passwordCreatedDesc', defaultMessage: 'You can enjoy your Password Manager' }) :
   intl.formatMessage({ id: 'passwordErrorDesc', defaultMessage: 'We could not modified your Master Password, try again later' })

  const nextButtonTextId = status === statusConst.SUCCESS ?  'access' : 'backToManager'

  const handleClickNext = () => {
    if(status === statusConst.SUCCESS){
      setCurrentStep(0)
    }else{
      handlePrevStep()
    }
  }

  return  (
    <>
      <Flex justifyContent='center' py={24}>
        <Box w={{xs: '100%', md: '50%'}}>
          <StatusCard variant={status} title={title} body={body}/>
        </Box>
      </Flex>
      <WizardFooter
        shouldHideBackButton={true} n
        nextButtonTextId={nextButtonTextId}
        onClickNext={handleClickNext}/>
    </>
  )
}

ResponseStatus.propTypes = {
  status: PropTypes.string
}


export default ResponseStatus