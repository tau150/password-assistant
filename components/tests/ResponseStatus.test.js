import * as React from 'react'
import { screen, waitFor } from '@testing-library/react'
import { act } from 'react-dom/test-utils';
import { render } from '../../test-helpers'
import userEvent from '@testing-library/user-event'
import { StepWizardContext } from '../../contexts/stepWizardContext';


import ResponseStatus from '../ResponseStatus'


const mockHandleNextStep = jest.fn()
const mockHandlePrevStep = jest.fn()
const mockSetCurrentStep = jest.fn()

const defaultContextProps = {
  currentStep: 0,
  setCurrentStep: mockSetCurrentStep,
  handleNextStep: mockHandleNextStep,
  handlePrevStep: mockHandlePrevStep,
  isStepDisabled: false
}

function WizardContextWrapper(props) {
  return (
    <StepWizardContext.Provider value={{...defaultContextProps}}>
      <ResponseStatus {...props} />
    </StepWizardContext.Provider>
  )
}


describe('Response Status test', () => {
  it('renders Response Status', async () => {
    expect(render(<WizardContextWrapper/>)).toMatchSnapshot()
  })

  it('should go to first step when status is success', async () => {
    render(<WizardContextWrapper status='success'/>)

    const nextButton = screen.getByRole('button', {name: 'next-button'})
    act( () => {
      userEvent.click(nextButton)
    });

    await waitFor(() => {
      expect(mockSetCurrentStep).toHaveBeenCalledWith(0)
    })
  })

  it('should go to prev step when status is not success', async () => {
    render(<WizardContextWrapper status='error'/>)

    const nextButton = screen.getByRole('button', {name: 'next-button'})
    act( () => {
      userEvent.click(nextButton)
    });

    await waitFor(() => {
      expect(mockHandlePrevStep).toHaveBeenCalled()
    })
  })

})

