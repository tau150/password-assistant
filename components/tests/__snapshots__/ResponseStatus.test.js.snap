// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response Status test renders Response Status 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-thupkg"
      >
        <div
          class="css-13l47xf"
        >
          <div
            class="css-dq00q9"
          >
            <svg
              class="chakra-icon css-qelcgd"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <div
              class="css-1t4in65"
            >
              <h4
                class="chakra-heading css-7n4zlk"
              >
                There was an error
              </h4>
              <p
                class="chakra-text css-13wg21h"
              >
                We could not modified your Master Password, try again later
              </p>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="css-teiq65"
      >
        <button
          aria-label="next-button"
          class="chakra-button css-1on1g6e"
          type="button"
        >
          Back to Password Manager
        </button>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-thupkg"
    >
      <div
        class="css-13l47xf"
      >
        <div
          class="css-dq00q9"
        >
          <svg
            class="chakra-icon css-qelcgd"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              stroke-width="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                stroke-linecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                stroke-linecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                stroke-miterlimit="10"
              />
            </g>
          </svg>
          <div
            class="css-1t4in65"
          >
            <h4
              class="chakra-heading css-7n4zlk"
            >
              There was an error
            </h4>
            <p
              class="chakra-text css-13wg21h"
            >
              We could not modified your Master Password, try again later
            </p>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="css-teiq65"
    >
      <button
        aria-label="next-button"
        class="chakra-button css-1on1g6e"
        type="button"
      >
        Back to Password Manager
      </button>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
